src/dbnode/encoding/proto/README.md

```diff
@@ -0,0 +1,241 @@
+
+### LRU Dictionary Compression
+
+The LRU dictionary compression scheme provides high levels of compression for `bytes` and `string` fields that match any of the following criteria:
+
+1. The value of the field changes infrequently.
+2. The value of the field changes frequently, but tends to rotate among a small number of frequently used values which may evolve over time.
+
+Similar to `LZ77` and its variants, this compression strategy has an implicit assumption that patterns in the input data occur close together. Data streams that don't satisfy this assumption will compress poorly.
+
+In the future, we may replace this simple algorithm with a more sophisticated dictionary compression scheme such as `LZ77`, `LZ78`, `LZW` (and its variants like `LZMW` and `LZAP`).
+
+#### Algorithm
+
+The encoder maintains a list of recently encoded strings in a per-field LRU. Everytime the encoder is about to encode a new string it first checks if the string is in the LRU.
+
+If the string *is not* in the LRU, then it encodes the string in its entirety and adds it to the LRU (evicting the least recently encoded string if necessary).
+
+If the string *is* in the LRU, then it encodes the **index** of the string in the LRU which requires much less space. For example, it only takes 2 bits to encode the position of the string in an LRU with a maximum capacity of 4 strings.
+
+Imagine compressing the following sequence of strings: `["foo", "bar", "baz", "bar"]` with an LRU of size 2. The steps below describe what is written into the stream, as well as what the state of the LRU is at the end:
+
+1. Write the full string "foo" into the stream and add "foo" to the LRU -> `["foo"]`
+2. Write the full string "bar" into the stream and add "bar" to the LRU -> `["foo", "bar"]`
+3. Write the full string "baz" into the stream and and evict "foo" from the LRU -> `["bar", "baz"]`
+4. Encode index 0 into the stream with a single bit (representing the string "bar") and update the LRU -> `["baz", "bar"]`
+
+This compression scheme works because the decoder can maintain its own LRU (of the same maximum capacity) and apply the same operations in the same order when its decompressing the stream. As a result, when it encounters an encoded LRU index it can lookup the corresponding string in its own LRU at the specified index.
+
+##### Encoding
+
+The LRU dictionary compression scheme uses 3 control bits to encode all the relevant information required to decode the stream.
+
+The first control bit is the "no change" control bit. If this is set to `1` then the value has not changed since its previous value and no further encoding/decoding is require.
+
+The second control bit determines how we will interpret the subsequent bits. If its `0` then we interpret the next `n` bits (where `n` is the number of bits required to encode the highest index of the configured LRU capacity) as the index into the LRU of recently encoded / decoded strings which this value corresponds to.
```

Comment:
nits: 
prefer `it is` to `its` here
prefer `remaining bits` over `subsequent bits`, to match the next paragraph
rather than describing `n` here, maybe add it to the first paragraph? i.e.
`The LRU dictionary compression scheme uses 2 control bits to encode all the relevant information required to decode the stream. Given a configured LRU cache which can be encoded in 'n' bytes:`